pkg/app/api/grpcapi/web_api.go

```diff
@@ -532,6 +534,18 @@ func (a *WebAPI) ListPipeds(ctx context.Context, req *webservice.ListPipedsReque
 		return nil, status.Error(codes.Internal, "Failed to get pipeds")
 	}
 
+	// Check piped connection status if necessary.
+	// The connection status of piped determined by its submitted stat in pipedStatCache.
+	if req.WithStatus {
+		for i := range pipeds {
+			if _, err := a.pipedStatCache.Get(pipeds[i].Id); err != nil {
+				pipeds[i].Status = model.Piped_OFFLINE
+				continue
+			}
```

Comment:
It should be OFFLINE only when the error is not-found, otherwise UNKNOWN is more appropriate.

Reply:
sure, addressed by https://github.com/pipe-cd/pipe/pull/2895/commits/2cd4e86a49fadcccee84eba0e511bc5bc0f6305b